### PR TITLE
vscode, vscodium: 1.45.1 -> 1.46.0

### PR DESCRIPTION
**Pull request by kuznero**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kuznero** [comment]
Does it make sense to introduce a small script to automatically update versions? If so, what tech is considered OK for this task (such that perhaps later on it is possible to automate on a bigger scale in NixPkgs)? ping @Mic92 .

**Comment by Synthetica9** [comment]
> Does it make sense to introduce a small script to automatically update versions? If so, what tech is considered OK for this task (such that perhaps later on it is possible to automate on a bigger scale in NixPkgs)? ping @Mic92 .

I don't think there is standardised tech for this; perhaps something to integrate it with nixpkgs-auto-upgrade would be nice. An "update-script" meta field?

**Comment by kuznero** [comment]
Everything should be ready, please merge.

**Comment by kuznero** [comment]
Now, also includes a bash script to fetch latest versions for both VSCode and VSCodium. Please review and merge.

File: pkgs/applications/editors/vscode/update.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+command -v grep >/dev/null 2>&1 || { echo >&2 "[grep] is required, but not installed.  Aborting."; exit 1; }
```
Comment (zowoq) [inline pkgs/applications/editors/vscode/update.sh:3]: Deps can be included by using `nix-shell`. 

Example: https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/minecraft/update.sh

**Review by zowoq — COMMENTED** [review]
I'd suggest moving the update script to a separate PR so that it doesn't block the version bumps.

File: pkgs/applications/editors/vscode/update.sh (hunk 1 above)
Comment (kuznero) [reply pkgs/applications/editors/vscode/update.sh]: Fixed in #90224

**Comment by zowoq** [comment]
Probably need to rebase this to drop the update script commits entirely rather than reverting.

**Comment by kuznero** [comment]
> Probably need to rebase this to drop the update script commits entirely rather than reverting.

Done

**Comment by zowoq** [comment]
@ofborg eval

**Review by zowoq — APPROVED** [review]
Checksums match.

**Comment by jonringer** [comment]
@GrahamcOfBorg build vscode
@GrahamcOfBorg build vscodium

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
[10 built, 24 copied (75.1 MiB), 11.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/90125
3 packages built:
vscode vscode-with-extensions vscodium
```

**Pull request merged**